Yoast/YoastSEO.js
# Implement TreeAssessor

## Summary

This PR can be summarized in the following changelog entry:

* _[not-user-facing]_ Adds the `TreeAssessor`, which is going to be used for assessments in the (near) future.

## Relevant technical choices:

* Adds a test assessment and test aggregator (in `specs/specHelpers`) to help with testing the `TreeAssessor` functionality.
* Fixes the JSDoc module structure for `tree/research` to include the `TreeResearcher` and `Research` classes.

## Test instructions

This PR can be tested by following these steps:

* Run the tests using `yarn test`. Check that the coverage of all files in the `src/tree/assess` folder is 100% (instead of `index.js`).
* Review the code. Check if you do not see any (edge) cases not covered by the implementation of the `TreeAssessor` and related functionality.

Fixes #2123
